source/common/http/http2/codec_impl.cc

```diff
@@ -569,7 +592,7 @@ int ConnectionImpl::onFrameReceived(const nghttp2_frame* frame) {
           // Even if we have :status 100 in the client case in a response, when
           // we received a 1xx to start out with, nghttp2 message checking
           // guarantees proper flow here.
-          stream->decodeHeaders();
+          stream->decodeHeaders(frame->headers.cat);
```

Comment:
I think this PR causes the following scenario handling change (not sure if it's intended):
Note that the line before this code (https://github.com/envoyproxy/envoy/blob/5c76fa31668b9b664c80ddc1b8aa659598d0d243/source/common/http/http2/codec_impl.cc#L590) sets `stream->waiting_for_non_informational_headers_ = false;`
In case this code handles a client 1xx response, IIUC the new code (this PR) will change that flag to true and continue execution, but the previous code seems to be keeping it false.
Is this new behavior desired?

Reply:
This is related to @alyssawilk comment above about naming and clarity. In this case the category is not HCAT_RESPONSE so it won't become true again. I will make this more clear.